nex/runner.go

```diff
@@ -84,26 +89,47 @@ func RunWorkload(ctx context.Context, logger *slog.Logger) error {
 		return errors.New("cannot start a function-type workload without specifying at least one trigger subject")
 	}
 
+	js, err := nc.JetStream()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed resolve jetstream: %s", err)
+		return err
+	}
+
+	bucket, err := js.ObjectStore(RunOpts.WorkloadURL.Hostname())
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed resolve workload object store: %s", err)
+		return err
+	}
+
+	artifact := RunOpts.WorkloadURL.Path[1:len(RunOpts.WorkloadURL.Path)]
+	fmt.Fprintf(os.Stdout, "resolved artifact name from workload url: %s", artifact)
+
+	info, err := bucket.GetInfo(artifact)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed resolve workload artifact: %s; %s", artifact, err)
+		return err
+	}
+
 	argv := []string{}
 	if len(RunOpts.Argv) > 0 {
 		argv = strings.Split(RunOpts.Argv, " ")
 	}
 
 	request, err := controlapi.NewDeployRequest(
 		controlapi.Argv(argv),
-		controlapi.Location(RunOpts.WorkloadUrl.String()),
+		controlapi.Hash(info.Digest),
```

Comment:
Note that this hash format is not the sanitized one that we like... it contains the `SHA-256:` prefix and I think it might even be in a different base. Might be worth running this through the hash "cleaner" that I think we used somewhere else (possibly in the CLI?)